mbed_app.json

```diff
@@ -41,9 +42,7 @@
             "mcuboot.slot-size": "0xC0000",
             "mcuboot.scratch-address": "0x80E0000",
             "mcuboot.scratch-size": "0x20000",
-            "mcuboot.max-img-sectors": "0x180",
-            "mcuboot.read-granularity": 1,
-            "qspif.QSPI_MIN_PROG_SIZE": 1
```

Comment:
There doesn't seem to be a default for this in [targets.json](https://github.com/ARMmbed/mbed-os/blob/f6e165febec105c042137a4d1bf663a58968aac8/targets/targets.json#L3753). Could the absence of this cause `bd` r/w to fail?

Reply:
The default for all targets is defined to be 1 here: https://github.com/ARMmbed/mbed-os/blob/f6e165febec105c042137a4d1bf663a58968aac8/storage/blockdevice/COMPONENT_QSPIF/mbed_lib.json#L21
So DISCO_L475VG_IOT01A has it set to 1.

Also, we can remove the override for EP_AGORA and NRF52840_DK too, because they both inherit MCU_NRF52840 which is already set to 4: https://github.com/ARMmbed/mbed-os/blob/f6e165febec105c042137a4d1bf663a58968aac8/storage/blockdevice/COMPONENT_QSPIF/mbed_lib.json#L40

Reply:
Added a commit to further clean up this for MCU_NRF52840